scene/gui/slider.cpp

```diff
@@ -269,98 +269,121 @@ void Slider::_notification(int p_what) {
 
 		case NOTIFICATION_DRAW: {
 			RID ci = get_canvas_item();
-			Size2i size = get_size();
-			double ratio = Math::is_nan(get_as_ratio()) ? 0 : get_as_ratio();
+			const Size2i size = get_size();
+			const double ratio = Math::is_nan(get_as_ratio()) ? 0 : get_as_ratio();
 
 			Ref<StyleBox> style = theme_cache.slider_style;
 			Ref<Texture2D> tick = theme_cache.tick_icon;
 
-			bool highlighted = editable && (mouse_inside || has_focus());
 			Ref<Texture2D> grabber;
-			if (editable) {
-				if (highlighted) {
-					grabber = theme_cache.grabber_hl_icon;
-				} else {
-					grabber = theme_cache.grabber_icon;
-				}
-			} else {
-				grabber = theme_cache.grabber_disabled_icon;
-			}
-
 			Ref<StyleBox> grabber_area;
-			if (highlighted) {
+
+			if (editable && (mouse_inside || has_focus())) {
+				grabber = theme_cache.grabber_hl_icon;
 				grabber_area = theme_cache.grabber_area_hl_style;
 			} else {
+				grabber = editable ? theme_cache.grabber_icon : theme_cache.grabber_disabled_icon;
 				grabber_area = theme_cache.grabber_area_style;
 			}
 
 			if (orientation == VERTICAL) {
-				int widget_width = style->get_minimum_size().width;
-				double areasize = size.height - (theme_cache.center_grabber ? 0 : grabber->get_height());
-				int grabber_shift = theme_cache.center_grabber ? grabber->get_height() / 2 : 0;
+				const int widget_width = style->get_minimum_size().width;
+				const double areasize = size.height - (theme_cache.center_grabber ? 0 : grabber->get_height());
+				const int grabber_shift = theme_cache.center_grabber ? grabber->get_height() / 2 : 0;
 				style->draw(ci, Rect2i(Point2i(size.width / 2 - widget_width / 2, 0), Size2i(widget_width, size.height)));
-				grabber_area->draw(ci, Rect2i(Point2i((size.width - widget_width) / 2, Math::round(size.height - areasize * ratio - grabber->get_height() / 2 + grabber_shift)), Size2i(widget_width, Math::round(areasize * ratio + grabber->get_height() / 2 - grabber_shift))));
+
+				if (symmetric_fill && get_min() < 0.0 && get_max() > 0.0 && !Math::is_equal_approx(get_min(), get_max())) {
```

Comment:
Just like with the warning, I don't think the last check is relevant (unless you consider min/max being very close, but not equal to zero, if that can even happen).

Reply:
We shouldn't call this method when `get_min()` and `get_max()` are approximately equal zero and the previous two checks will only allow this case which can cause division of zero crash.

Reply:
Maybe we can use `!Math::is_zero_approx(get_max() + get_min())`.

Reply:
Eh, the new condition is basically the same.

Reply:
Without this condition `!Math::is_zero_approx(get_min() - get_max())` the next line may crash 
`const double zero_ratio = get_min() / (get_min() - get_max());`, so we switch to normal fill instead.

Reply:
Removed it by using CMP_EPSILON instead of 0, and rebased.
`if (symmetric_fill && get_min() < -CMP_EPSILON && get_max() > CMP_EPSILON)`